cupy/cuda/cutensor.pyx

```diff
@@ -171,39 +223,33 @@ cpdef inline check_status(int status):
 
 
 ###############################################################################
-# Handler creation/destruction
+# Handler initialization
 ###############################################################################
 
-cpdef size_t create() except? 0:
-    """Initializes the cuTensor library"""
-    cdef Handle handle
-    with nogil:
-        status = cutensorCreate(&handle)
-    check_status(status)
-    return <size_t>handle
-
-
-cpdef destroy(size_t handle):
-    """Release hardware resources used by cuTensor library"""
+cpdef intptr_t init() except? 0:
+    """Initializes the cuTENSOR library"""
+    cdef Handle *handle = <Handle*>PyMem_Malloc(sizeof(Handle))
     with nogil:
-        status = cutensorDestroy(<Handle>handle)
+        status = cutensorInit(handle)
     check_status(status)
+    return <intptr_t>handle
```

Comment:
It seems this handle never gets freed? Perhaps you need to register it somewhere for cleanup?

Reply:
How about RAII way I suggested above (https://github.com/cupy/cupy/pull/2709#discussion_r350770416)?

Reply:
I think `cupy/cuda/cutensor.pyx` should be just a **thin** wrapper of cuTENSOR APIs as much as possible. So, what about wrapping a handle with its de-constructor in `cupy/cutensor.py` so that it can be freed?

Reply:
I think this is the point careful consideration is needed.

In general resource acquisition and release should be symmetric.
If one module allocates a memory, it's that module that should release it.

So, if we want to leave `cutensor.pyx` really thin, memory allocation should also be done outside.
That's indeed an option.

Another more convenient option is what I suggested.

Reply:
We are still discussing what is the ideal interface. Considering this is currently non-public API (we can refactor later), we will proceed to merge this now.